lambda/lambda.ml

```diff
@@ -48,6 +52,7 @@ type primitive =
   | Psetglobal of Ident.t
   (* Operations on heap blocks *)
   | Pmakeblock of int * mutable_flag * block_shape
```

Comment:
Do we assert somewhere that `Pmakeblock` is not used incorrectly with `lazy_tag`/`forward_tag`?

Reply:
No. I did check all occurrences of `lazy_tag` in the compiler's code base, but if some user writes `Obj.with_tag 246 (Obj.repr { contents = foo })`, this will turn into a `Pmakeblock` with tag `Lazy_tag` and not a `Pmakelazyblock`.
If we wanted to be more robust, we could insert some code in `Simplif` to turn all `Pmakeblock` with a lazy-related tag into `Pmakelazyblock`, but I don't think it's worth the trouble.